Update documentation to include Ceilometer metrics to the discussion (#130)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/proc_configuring-red-hat-openstack-platform-overcloud-for-stf.adoc

. Create a configuration file called `stf-connectors.yaml` in the `/home/stack` directory.
+

> IMPORTANT: The Service Telemetry Operator simplifies the deployment of all data ingestion and data storage components for single cloud deployments. To share the data storage domain with multiple clouds, see xref:configuring-multiple-clouds_advanced-features[]. Additionally, setting `EventPipelinePublishers` and `MetricPipelinePublishers` to empty lists will result in no metric or event data passing to {OpenStack} legacy telemetry components, such as Gnocchi or Panko. If you require sending to additional pipelines, the Ceilometer polling interval of 5 seconds as specified in `ExtraConfig` will likely overwhelm the legacy components. If a longer polling interval is configured, then {ProjectShort} will need to also be modified in order to avoid stale metrics, resulting in what appears to be missing data in Prometheus.

Reviewer: ```suggestion
IMPORTANT: The Service Telemetry Operator simplifies the deployment of all data ingestion and data storage components for single cloud deployments. To share the data storage domain with multiple clouds, see xref:configuring-multiple-clouds_advanced-features[]. Additionally, setting `EventPipelinePublishers` and `MetricPipelinePublishers` to empty lists results in no metric or event data passing to {OpenStack} legacy telemetry components, such as Gnocchi or Panko. If you you to sent data to additional pipelines, the Ceilometer polling interval of 5 seconds as specified in `ExtraConfig` might overwhelm the legacy components. If you configure a longer polling interval, you must also modify {ProjectShort} to avoid stale metrics, resulting in what appears to be missing data in Prometheus.
```

IMPORTANT: The Service Telemetry Operator simplifies the deployment of all data ingestion and data storage components for single cloud deployments. To share the data storage domain with multiple clouds, see xref:configuring-multiple-clouds_advanced-features[].

. In the `stf-connectors.yaml` file, configure the `MetricsQdrConnectors` address to connect the {MessageBus} on the overcloud to the {ProjectShort} deployment.
* Add the `CeilometerQdrPublishMetrics: true` parameter to enable collection and transport of Ceilometer metrics to {ProjectShort}.
* Add the `CeilometerQdrPublishEvents: true` parameter to enable collection and transport of Ceilometer events to {ProjectShort}.

> * Add the `EventPiplinePublishers: []` and `MetricPipelinePublishers: []` to avoid writing data to Gnocchi and Panko.

Reviewer 2: That may be not desired e.g for autoscaling you definitely want data in both locations.

Author: Correct, but that will need to be covered in a separate part of the document. I am expecting only STF to be used for monitoring of OSP.

I do think we should probably add a note or warning though here.

Author: Added to the IMPORTANT section leading into this conversation. Let me know if that addresses the concerns.

* Add the `ManagePolling: true` and `ManagePipeline: true` parameters to allow full control of Ceilometer polling and pipeline configuration.

> * Add the `ExtraConfig` parameter `ceilometer::agent::polling::polling_interval` to set the polling interval of Ceilometer to be compatible with the default {ProjectShort} scrape interval.

Reviewer 2: Note on low polling interval (low values) if not storing data in gnocchi, high interval (with values like 300 or 600s) if also storing data in gnocchi.

Author: Yea I'll add another note here as well. The expectation is that there is no gnocchi or panko endpoints. If there are (and a lower polling resolution is desired) then we'll need to address that on the STF side of things by adjusting the default polling interval for ceilometer gateways. I expect more documentation to need to be written for this scenario...

Author: Used the same IMPORTANT note section. Let me know if this is addresses. Thanks!
